src/components/MasterItemsSelect.js

```diff
@@ -1,25 +1,61 @@
 import { Component, PropTypes, DOM as dom } from 'react';
+import _ from 'underscore';
 
 class MasterItemsSelect extends Component {
   constructor(props) {
     super(props);
     this.handleChange = this.handleChange.bind(this);
+    this.handleChangeCategory = this.handleChangeCategory.bind(this);
+    this.state = {
+      filteredItems: this.filterItems(this.props.masterItems[0].category)
```

Comment:
I think this is fine. It does always feel off to me to store an entire array in local state, but I can't think of a better way to do it..